addons/mrp/models/mrp_production.py

```diff
@@ -687,19 +687,15 @@ def _onchange_producing(self):
 
     @api.onchange('lot_producing_id')
     def _onchange_lot_producing(self):
-        if self.product_id.tracking == 'serial':
-            if self.env['stock.move.line'].search_count([
-                ('company_id', '=', self.company_id.id),
-                ('product_id', '=', self.product_id.id),
-                ('lot_id', '=', self.lot_producing_id.id),
-                ('state', '!=', 'cancel')
-            ]):
-                return {
-                    'warning': {
-                        'title': _('Warning'),
-                        'message': _('Existing Serial number (%s). Please correct the serial numbers encoded.') % self.lot_producing_id.name
-                    }
+        try:
+            self._check_sn_uniqueness()
```

Comment:
I think it's a bit strange to have a usererror pop up about a byproduct's sn already being used when you change the production's sn. Also, your solution will remove the check that the SN doesn't already exist in stock even when it wasn't manufactured (i.e. Product form > Update Quantity > create a SN, then create a MO for the product > use the SN you created in Update Quantity).

It also causes some weirdness when changing the SN in an already done MO because it will check the existing SN, not the new SN that you are typing in.

It's probably better to leave this independent of `_check_sn_uniqueness` and maybe create a new private method that is used by both `_onchange_lot_producing` and `_check_sn_uniqueness` to check only the manufactured product's SN uniqueness avoid repeated code.